jobs/debug-pod.Jenkinsfile

```diff
@@ -0,0 +1,187 @@
+             defaultValue: "",
+             trim: true),
+      string(name: 'TIMEOUT',
+             description: 'Timeout value',
+             defaultValue: "1h",
+             trim: true),             
+
+    ]),
+    durabilityHint('PERFORMANCE_OPTIMIZED')
+])
+
+def build_description = "[${params.STREAM}][${params.ARCH}]"
+
+// Reload pipecfg if a hotfix build was provided. The reason we do this here
+// instead of loading the right one upfront is so that we don't modify the
+// parameter definitions above and their default values.
+if (params.PIPECFG_HOTFIX_REPO || params.PIPECFG_HOTFIX_REF) {
+    node {
+        pipecfg = pipeutils.load_pipecfg(params.PIPECFG_HOTFIX_REPO, params.PIPECFG_HOTFIX_REF)
+        build_description = "[${params.STREAM}-${pipecfg.hotfix.name}][${params.ARCH}]"
+    }
+}
+
+// runtime parameter always wins
+def cosa_img = params.COREOS_ASSEMBLER_IMAGE
+cosa_img = cosa_img ?: pipeutils.get_cosa_img(pipecfg, params.STREAM)
+
+def stream_info = pipecfg.streams[params.STREAM]
+
+def cosa_controller_img = stream_info.cosa_controller_img_hack ?: cosa_img
+cosa_controller_img = 'quay.io/dustymabe/coreos-assembler-staging:cosa-tmux'
+// DELETE ^^
+
+// If we are a mechanical stream then we can pin packages but we
+// don't maintain complete lockfiles so we can't build in strict mode.
+def strict_build_param = stream_info.type == "mechanical" ? "" : "--strict"
+
+// Note that the heavy lifting is done on a remote node via podman
+// --remote so we shouldn't need much memory.
+def cosa_memory_request_mb = 512
+
+// the build-arch pod is mostly triggering the work on a remote node, so we
+// can be conservative with our request
+def ncpus = 1
+
+echo "Waiting for build-${params.STREAM}-${params.ARCH} lock"
+currentBuild.description = "${build_description} Waiting"
+
+// declare these early so we can use them in `finally` block
+assert params.VERSION != ""
+def newBuildID = params.VERSION
+def basearch = params.ARCH
+
+// matches between build/build-arch job
+def timeout_mins = 240
+
+//def userName = currentBuild.getBuildCauses()[-1].getUser().getFullName()
+def userName = currentBuild.getBuildCauses('hudson.model.Cause$UserIdCause')[0]?.userName ?: 'default'
+println "Jenkins username: ${userName}"
+
+
+// release lock: we want to block the release job until we're done.
+// ideally we'd lock this from the main pipeline and have lock ownership
+// transferred to us when we're triggered. in practice, it's very unlikely the
+// release job would win this race.
+lock(resource: "release-${params.VERSION}-${basearch}") {
+// build lock: we don't want multiple concurrent builds for the same stream and
+// arch (though this should work fine in theory)
+lock(resource: "build-${params.STREAM}-${basearch}") {
+    cosaPod(cpu: "${ncpus}",
+            memory: "${cosa_memory_request_mb}Mi",
+            image: cosa_controller_img,
+            serviceAccount: "jenkins" ,
+            user : "${userName}") {
```

Comment:
for this argument to `cosaPod` I would just name the argument `name`, as in the name of the pod. i.e. your coreos-ci-lib PR would be to add a `name` argument to `pod.groovy`.

Reply:
I have renamed the "user" to be "name"